ste/JobPartPlanFileName.go

```diff
@@ -214,42 +215,74 @@ func (jpfn JobPartPlanFileName) Create(order common.CopyJobPartOrderRequest) {
 	// Initialize the offset for the 1st transfer's src/dst strings
 	currentSrcStringOffset := eof + int64(unsafe.Sizeof(JobPartPlanTransfer{}))*int64(jpph.NumTransfers)
 
+	// Determine the maximum destination length:
+	var maxDestLength uint32 = math.MaxUint32
+
+	switch order.FromTo.To() {
+	case common.ELocation.Blob():
+		// The maximum length is 1024 characters
+		maxDestLength = 1024
+	case common.ELocation.File():
+		// it's actually 255 per segment, for a total of 2048
+		maxDestLength = 2048
+	case common.ELocation.BlobFS():
+		maxDestLength = 1024
+		// Leave it as is if it's a default.
+	}
+
 	// Write each transfer to the Job Part Plan file (except for the src/dst strings; comes come later)
 	for t := range order.Transfers {
-		if len(order.Transfers[t].Source) > math.MaxInt16 || len(order.Transfers[t].Destination) > math.MaxInt16 {
-			panic(fmt.Sprintf("The file %s exceeds azcopy's current maximum path length on either the source or the destination.", order.Transfers[t].Source))
+		if uint32(len(order.Transfers[t].Source)) > uint32(math.MaxUint32) || uint32(len(order.Transfers[t].Destination)) > maxDestLength {
```

Comment:
Remember, if we change _anything_ in the plan files, we must always change const DataSchemaVersion common.Version

Reply:
I'd assumed we'd already changed it this revision-- I'll increment it.